core/iwasm/common/wasm_runtime_common.c

```diff
@@ -3929,11 +3937,108 @@ wasm_runtime_get_export_type(WASMModuleCommon *const module, int32 export_index,
         const WASMExport *wasm_export = &wasm_module->exports[export_index];
         export_type->name = wasm_export->name;
         export_type->kind = wasm_export->kind;
+        if (wasm_export->kind == EXPORT_KIND_FUNC) {
+            export_type->u.func_type =
+                wasm_module
+                    ->functions[wasm_export->index
+                                - wasm_module->import_function_count]
+                    ->func_type;
+        }
         return;
     }
 #endif
 }
 
+int32
+wasm_func_type_get_param_count(WASMFuncType *const func_type)
+{
+    if (!func_type) {
+        bh_assert(0);
+        return -1;
+    }
+
+    return func_type->param_count;
+}
+
+wasm_valkind_t
+wasm_func_type_get_param_valkind(WASMFuncType *const func_type,
+                                 uint32 param_index)
+{
+    if (!func_type || (param_index >= func_type->param_count)) {
+        bh_assert(0);
+        return (wasm_valkind_t)-1;
+    }
+
+    switch (func_type->types[param_index]) {
+        case VALUE_TYPE_I32:
+            return WASM_I32;
+        case VALUE_TYPE_I64:
+            return WASM_I64;
+        case VALUE_TYPE_F32:
+            return WASM_F32;
+        case VALUE_TYPE_F64:
+            return WASM_F64;
+        case VALUE_TYPE_FUNCREF:
+            return WASM_FUNCREF;
+
+        case VALUE_TYPE_V128:
+        case VALUE_TYPE_EXTERNREF:
```

Comment:
The latest wasm-c-api wasm.h has changed wasm_valkind_enum to:
```C
enum wasm_valkind_enum {
  WASM_I32,
  WASM_I64,
  WASM_F32,
  WASM_F64,
  WASM_EXTERNREF = 128,
  WASM_FUNCREF,
};
```
https://github.com/WebAssembly/wasm-c-api/blob/2ce1367c9d1271c83fb63bef26d896a2f290cd23/include/wasm.h#L180-L187

We had better align to it and also support SIMD VALUE_TYPE_V128, so I think we had better change it to below in both wasm-c-api.h and wasm_export.h:
```C
enum wasm_valkind_enum {
  WASM_I32,
  WASM_I64,
  WASM_F32,
  WASM_F64,
  WASM_V128,
  WASM_VOID,
  WASM_EXTERNREF = 128,
  WASM_FUNCREF,
};
```
There may be many places to change WASM_ANYREF to WASM_EXTERNREF inside runtime src and samples, it is up to you to decide whether to change it or not. But here had better also handle VALUE_TYPE_V128 and VALUE_TYPE_EXTERNREF:
```C
#if WASM_ENABLE_SIMD != 0
    case VALUE_TYPE_V128:
        return WASM_V128;
#endif
#if WASM_ENABLE_REF_TYPES != 0
    case VALUE_TYPE_EXTERNREF:
        return WASM_EXTERNREF;  (or WASM_ANYREF if it isn't changed)
#endif
```

Reply:
ps. no need to add `WASM_VOID` to `wasm_valkind_enum`, it isn't used